src/SFA.DAS.AssessorService.Application/Handlers/Settings/SetSettingHandler.cs

```diff
@@ -15,10 +15,30 @@ public SetSettingHandler(ISettingRepository settingRepository)
             _settingRepository = settingRepository;
         }
 
-        public async Task<Unit> Handle(SetSettingRequest request, CancellationToken cancellationToken)
+        public async Task<SetSettingResult> Handle(SetSettingRequest request, CancellationToken cancellationToken)
         {
-            await _settingRepository.SetSetting(request.Name, request.Value);
-            return Unit.Value;
+            var result = new SetSettingResult();
+            if(request.Name.Length == 0 || request.Name.Length > 50)
```

Comment:
The usual pattern is to have a separate validator to validate the request.

Reply:
Ok - I see that now